luigi_pipeline/lib/model/gcnv_mt_schema.py

```diff
@@ -0,0 +1,212 @@
+        return self.mt.start
+
+    @row_annotation(fn_require=[contig, pos])
+    def xpos(self):
+        return variant_id.get_expr_for_xpos(
+            hl.locus(self.mt.contig, self.mt.pos)
+        )
+
+    @row_annotation(fn_require=xpos)
+    def xstart(self):
+        return self.mt.xpos
+
+    @row_annotation(fn_require=[contig, end])
+    def xstop(self):
+        return variant_id.get_expr_for_xpos(
+            hl.locus(self.mt.contig, self.mt.end)
+        )
+
+class SeqrGCNVGenotypesSchema(SeqrGenotypesSchema):
+
+    def __init__(self, *args, is_new_joint_call=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._is_new_joint_call = is_new_joint_call
+
+    @row_annotation(fn_require=SeqrGenotypesSchema.genotypes)
+    def samples(self):
+        return self._genotype_filter_samples(lambda g: True)
+
+    @row_annotation(fn_require=SeqrGenotypesSchema.genotypes)
+    def samples_new_call(self):
+        return self._genotype_filter_samples(lambda g: g.new_call)
+
+    def samples_no_call(self):
+        pass
+
+    def samples_num_alt(self):
+        pass
+
+    def samples_gq(self):
+        pass
+
+    def samples_ab(self):
+        pass
+
+    @row_annotation(fn_require=SeqrGenotypesSchema.genotypes)
+    def samples_qs(self, start=0, end=1000, step=10):
+        return hl.struct(**{
+            '%i_to_%i' % (i, i+step): self._genotype_filter_samples(lambda g: ((g.qs >= i) & (g.qs < i+step)))
+            for i in range(start, end, step)
+        }, **{
+            "samples_qs_gt_1000": self._genotype_filter_samples(lambda g: g.qs > 1000)
+        })
+
+    @row_annotation(name="samples_cn", fn_require=SeqrGenotypesSchema.genotypes)
+    def samples_cn(self, start=0, end=4, step=1):
+        return hl.struct(**{
+            '%i' % i: self._genotype_filter_samples(lambda g: g.cn == i)
+            for i in range(start, end, step)
+        }, **{
+            "samples_cn_gte_4": self._genotype_filter_samples(lambda g: g.cn >= 4)
+        })
+
+    def _genotype_filter_samples(self, filter):
+        samples = self.mt.genotypes.filter(filter).map(lambda g: g.sample_id)
+        return hl.if_else(hl.len(samples) > 0, samples, hl.missing(hl.dtype('array<str>')))
+    
+    def _genotype_fields(self):
+        return {
+            'sample_id': self.mt.sample_fix.first_match_in(SAMPLE_ID_REGEX)[0],
+            'qs': self.mt.QS,
+            'cn': self.mt.CN,
+            'defragged': self.mt.defragmented,
+            # Hail expression is to bool-ify a string value.
+            'prev_call': hl.if_else(hl.len(self.mt.identical_ovl) > 0, True, False) if self._is_new_joint_call else ~self.mt.is_latest,
+            'prev_overlap': hl.if_else(hl.len(self.mt.any_ovl) > 0, True, False)  if self._is_new_joint_call else False,
+            # NB: previous implementation also falsified NA, but hail treats NA as an empty value.
+            'new_call': self.mt.no_ovl if self._is_new_joint_call else False,
```

Comment:
I tried to default to hail's `NA` handling in this implementation... but the behavior might be slightly different on this one: https://github.com/broadinstitute/seqr-loading-pipelines/blob/master/sv_pipeline/exome/load_data.py#L132

Reply:
It might actually  be easier to read if you define a whole dictionary based on whether or not its a new join call, instead of having al the fields be `if else` inline.
 So something like
```
fields = {...}
if self._is_new_joint_call:
  fields.update({'prev_overlap': ....})
else:
  fields.update({'prev_overlap': ....})
```

Reply:
I don't think hail has any special `NA` handling here,  because theres an actual hail missing value and then theres a string that says "NA" and the latter is what we have in our data and hail will just treat that as a string

Reply:
Hail has been parsing `NA` to the actual hail missing value for us here... though as above... maybe we want to be defensive.

Reply:
I didn;tknow it did that, thats cool. I don't think we need to be defensive if so